sql-plugin/src/main/301db/scala/com/nvidia/spark/rapids/shims/v2/GpuBroadcastHashJoinExec.scala

```diff
@@ -42,16 +43,64 @@ class GpuBroadcastHashJoinMeta(
     join.leftKeys.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
   val rightKeys: Seq[BaseExprMeta[_]] =
     join.rightKeys.map(GpuOverrides.wrapExpr(_, conf, Some(this)))
-  val condition: Option[BaseExprMeta[_]] =
+  val conditionMeta: Option[BaseExprMeta[_]] =
```

Comment:
In order to replace this with a nested loop join, we need to fold in the "equality" condition into the "inequality" condition that Spark already separated when choosing a hash join.  The nested loop join needs the entire condition which is _not_ covered by the `condition` parameter in the join node in Spark.  This needs a separate condition, like nestedLoopCondition, where we "tack-on" the equality condition to the expression.  For example if the hash join is on keys X and Y with a condition of W > Z then the `condition` will be simply `W > Z` but the nested loop join needs it to be `X == Y AND W > Z`.  There are some types that could cause the equality itself to preclude being an AST (e.g.: AST cannot compare strings even for equality today).